src/v1/device_manager_client.ts

```diff
@@ -1042,23 +1042,50 @@ export class DeviceManagerClient {
       {} | undefined
     ]
   > | void {
-    request = request || {};
-    let options: CallOptions;
-    if (typeof optionsOrCallback === 'function' && callback === undefined) {
-      callback = optionsOrCallback;
-      options = {};
-    } else {
-      options = optionsOrCallback as CallOptions;
-    }
-    options = options || {};
-    options.otherArgs = options.otherArgs || {};
-    options.otherArgs.headers = options.otherArgs.headers || {};
-    options.otherArgs.headers['x-goog-request-params'] =
-      this._gaxModule.routingHeader.fromParams({
-        'device.name': request.device!.name || '',
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
+      const token_response = await this.getRegistryToken();
```

Comment:
@ronak-ingress @prashantdatirIngress we need to come up with a way so that we don't call getRegistryToken every time a method is called. We should ask users to do some initialization step which will call getRegistryToken and then store the token in the SDK. No need to address this in this PR, I'll create a task for it